src/sail_json_backend/json.ml

```diff
@@ -398,14 +398,115 @@ let remove_identity_funcs op =
       else op
   ) op identity_funcs
 
+let extract_captured_operand pattern s =
+  if Str.string_match pattern s 0 then
+    let captured_group = Str.matched_group 0 s in
+    let optional_operand_parts = String.split_on_char '(' captured_group in
+    let captured_op = List.hd optional_operand_parts in
+    debug_print ("Captured operand: " ^ captured_op);
+    Some captured_op
+  else
+    None
+
+let extract_opt_operands_fn pattern op_list =
+    (* 
+       This extracts the optional operands function names from the captured regex:
+       eg: maybe_vmask(vm) -> maybe_vmask
+       This is further used as key to the mappings table to extract the optional
+       operands.
+    *)
+  let captured_operand = List.find_map (fun s -> extract_captured_operand pattern s) op_list in
+  match captured_operand with
+  | Some op -> Some true, op
+  | None -> None, ""
+
+let optional_operand k op =
+    (*
+        The current regex pattern targets a specific function name pattern 
+        that starts with "maybe_*", such as "maybe_vmask", "maybe_lmul_flag", and "maybe_ta_flag".
+        Since those kinds are the most prevalent ones as of now.
+
+        TODO: come up with a more robust implementation, that doesn't involve
+        specifying function names explicitly.
+    *)
+  let pattern = Str.regexp_case_fold ("maybe[a-z0-9_]+\\(" ^ "(" ^ op ^ ")" ^ "\\)") in
+    match Hashtbl.find_opt assembly_clean k with
+    | Some op_list -> extract_opt_operands_fn pattern op_list
+    | None -> None, ""
+
+
+(*
+    This examines zero-bit patterns like 0b0, 0b00, 0b000, etc.,
+    retrieved from the mappings Hashtbl to deduce the default operands.
+
+    See:
+    https://github.com/ThinkOpenly/sail-riscv/blob/46750eed2796799fcc4bb220af8ce45ae9fee76c/
+    model/riscv_insts_vext_vset.sail#L51
+
```

Comment:
I have mixed feelings on this. Since you give a nice example below, I'm leaning toward removing this reference link.